### ISSUE TITLE
Send ancestors to droppable callback

### ISSUE THREAD

**Issue by jerryorr** [SpiderStrategies/kalpa-tree#371]
The `droppable` callback currently gets two arguments: the node being moved and the target parent. I also need all the ancestors of the target parent.

**Comment by mattsgarlata** [comment]
Couldn't you query the tree to discover the ancestors on your own?  That may be a better way to go about this.  Whose to say that we won't come up with some other piece of data we'll need in the future.

**Comment by mattsgarlata** [comment]
To elaborate, I believe each node in the tree has a parentId, and I think the tree has a get() method to retrieve a node in the tree, so you can traverse using the parentId.

**Comment by jerryorr** [comment]
Yes, that would work, but @nathanbowser thinks it will be more performant to send them at once rather than recursively calling `tree.parent()`

**Comment by mattsgarlata** [comment]
@nathanbowser are you sure this doesn't weaken the API?  What happens next time when we want some other weird data like the descendants or siblings or something?  I really am not convinced an API change is a good idea here.

Also, the deepest tree in all of SMS is less than 27 nodes deep.  So we're talking about a very cheap traversal here.  Much cheaper than traversing all the nodes in the tree, and an order of magnitude cheaper than drawing the tree.  So I don't think performance should be a concern here.

**Comment by jerryorr** [comment]
If we do decide not to send the ancestors, then it would be helpful to at least have the tree itself as an argument for the callback. The `droppable` callback is passed in as a constructor option, so I don't have a reference to the tree yet.

**Comment by mattsgarlata** [comment]
I think passing a reference to the tree is a great idea

**Comment by nathanbowser** [comment]
Wow, I thought I was using d3-hierarchy, which had some sweet new code in v4 for ancestors/descendants, but then I remembered I ditched that for our own layout code, which ended up being faster for us because we didn't need all the bloat, and far more simple. Dang.

Which means the node objects I store in `_layout` aren't d3.hierarchy nodes, so they don't have the built-in ancestors/descendants functions.

TL;DR: Ignore our previous conversation @jerryorr. We could add an ancestors/descendants function, but I think for simplicity I can just pass the tree reference.
